includes/forms/class-llms-forms.php

```diff
@@ -630,6 +630,85 @@ public function get_form_post( $location, $args = array() ) {
 
 	}
 
+	/**
+	 * Check whether a given form is a core form.
+	 *
+	 * When in presence of more than one form per location, the core form is identified as the one with lower id.
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_Post|int $form Form's WP_Post instance, or its ID.
+	 * @return boolean
+	 */
+	public function is_a_core_form( $form ) {
+
+		$form_id = $form instanceof WP_Post ? $form->ID : $form;
+
+		if ( ! $form_id ) {
+			return false;
+		}
+
+		return in_array( $form_id, $this->get_core_forms( 'ids' ), true );
+
+	}
+
+	/**
+	 * Retrieves only core forms.
+	 *
+	 * When in presence of more than one form per location it selects the one with lower id.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $return What to return: 'posts', for an array of WP_Post; 'ids' for an array of WP_Post ids.
+	 * @return WP_Post[]|int[]
+	 */
+	private function get_core_forms( $return = 'posts', $use_cache = true ) {
+
+		global $wpdb;
+
+		$forms_cache_key = 'posts' === $return ? 'llms_core_forms' : 'llms_core_form_ids';
+		$forms           = $use_cache ? wp_cache_get( $forms_cache_key ) : false;
+
+		if ( false !== $forms ) {
+			return $forms;
+		}
+
+		$locations              = array_keys( $this->get_locations() );
+		$locations_placeholders = implode( ',', array_fill( 0, count( $locations ), '%s' ) );
+		$prepare_values         = array_merge( $locations, array( $this->get_post_type() ) );
+
+		$query = "
+SELECT MIN({$wpdb->posts}.ID) AS ID
+FROM $wpdb->posts
+INNER JOIN {$wpdb->postmeta} ON ( {$wpdb->posts}.ID = {$wpdb->postmeta}.post_id )
+INNER JOIN {$wpdb->postmeta} AS mt1 ON ( {$wpdb->posts}.ID = mt1.post_id )
+WHERE 1=1
+AND (
+	( {$wpdb->postmeta}.meta_key='_llms_form_location' AND {$wpdb->postmeta}.meta_value IN ({$locations_placeholders}) )
+	AND
+	( mt1.meta_key='_llms_form_is_core' AND mt1.meta_value='yes' )
+	)
+AND {$wpdb->posts}.post_type=%s
+GROUP BY {$wpdb->postmeta}.meta_value";
+
+		$form_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				$query, // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- It is prepared.
+				$prepare_values
+			)
+		);
+
+		$form_ids = array_map( 'absint', $form_ids );
+		$forms    = 'post' === $return ? array_map( 'get_post', $form_ids ) : $form_ids;
+
+		// Cache anyways.
```

Comment:
Is this comment needed?
```suggestion
```

Reply:
Are comments _needed_? :D 
I added it to tell the reader "Hey I know what I'm doing, I want to cache the value anyways even if `$use_cache` is `false`".

Reply:
But at this point, we've already checked the cache. It was empty, so we ran the query and are adding it to the cache. There's no question about whether we should or shouldn't cache the result by the time we get to this line.